### PR TITLE
Mark dtype for array creation functions

### PR DESCRIPTION
**Pull request by prasunanand**
https://github.com/Quansight-Labs/unumpy/issues/7

### CODE REVIEW

File: unumpy/_multimethods.py [hunk 1]
```diff
@@ -324,33 +326,33 @@ def arange(start, stop=None, step=None, dtype=None):
 
 @create_numpy(_identity_argreplacer)
 def array(object, dtype=None, copy=True, order="K", subok=False, ndmin=0):
```
Comment (hameerabbasi) [inline unumpy/_multimethods.py:326]: There is already a replacer for this: `_dtype_argreplacer`. Don't change the identity argreplacer.

Comment (prasunanand) [reply unumpy/_multimethods.py:326]: Got it

Comment (prasunanand) [reply unumpy/_multimethods.py:326]: Made changes.

**Review by hameerabbasi — APPROVED** [review]

**Comment by hameerabbasi** [comment]
Thanks, @prasunanand!

**Pull request merged**